tests/unit/test_polynomialembeddor.py

```diff
@@ -0,0 +1,343 @@
+        eden_proc._linear = False
+
+    # def test_clique_12_uniformity(self):
+    #     eden_proc = self.eden_proc
+    #     cliques = defaultdict(int)
+    #     runs = 100
+    #     for _ in range(runs):
+    #         emb = eden_proc.largestNativeClique(max_chain_length=4)
+    #         cliques[frozenset(map(tuple, emb))] += 1
+    #     assert len(cliques) == 2, "should have found exactly 2 cliques, got %d."%len(cliques)
+    #     A, B = cliques.keys()
+    #     verify_clique(eden_proc, list(A), 12, 4)
+    #     verify_clique(eden_proc, list(B), 12, 4)
+
+    #     a, b = cliques.values()
+    #     assert abs(a - b) < 3 * (runs**.5), "%s and %s should be roughly equal.  This test has about a 2%% chance of failure due to random chance." % (a, b)
+    #     # shoddy statistics:
+    #     # >>> from random import randint
+    #     # >>> M = 100
+    #     # >>> N = 1000000
+    #     # >>> f = sum(abs(2*sum(randint(0,1) for _ in range(M))-M)>3*(M**.5) for _ in range(N))/float(N)
+    #     # >>> print "%.2f chance of failure"%f
+
+    def test_clique_12(self):
+        eden_proc = self.eden_proc
+        emb = eden_proc.largestNativeClique()
+        verify_clique(eden_proc, emb, 12, 4)
+
+    def test_biclique_balanced(self):
+        eden_proc = self.eden_proc
+        cliques = defaultdict(int)
+        emb = eden_proc.largestNativeBiClique(chain_imbalance=0)
+        verify_biclique(eden_proc, emb, 12, 9, 3, 3)
+
+    def test_biclique_balanced_length2(self):
+        eden_proc = self.eden_proc
+        cliques = defaultdict(int)
+        emb = eden_proc.largestNativeBiClique(
+            chain_imbalance=0, max_chain_length=2)
+        verify_biclique(eden_proc, emb, 8, 8, 2, 2)
+
+    def test_biclique_balanced_length3(self):
+        eden_proc = self.eden_proc
+        cliques = defaultdict(int)
+        emb = eden_proc.largestNativeBiClique(
+            chain_imbalance=None, max_chain_length=3)
+        verify_biclique(eden_proc, emb, 12, 9, 3, 3)
+
+    def test_biclique_largest(self):
+        eden_proc = self.eden_proc
+        cliques = defaultdict(int)
+        emb = eden_proc.largestNativeBiClique(chain_imbalance=None)
+        verify_biclique(eden_proc, emb, 16, 12, 3, 7)
+
+    def test_biclique_tightest(self):
+        eden_proc = self.eden_proc
+        cliques = defaultdict(int)
+        emb = eden_proc.tightestNativeBiClique(8, 12, chain_imbalance=None)
+        verify_biclique(eden_proc, emb, 8, 12, 3, 2)
+
+    def test_biclique_tightest88(self):
+        eden_proc = self.eden_proc
+        cliques = defaultdict(int)
+        emb = eden_proc.tightestNativeBiClique(8, chain_imbalance=None)
+        verify_biclique(eden_proc, emb, 8, 8, 2, 2)
+
+    def test_biclique_tightest(self):
+        eden_proc = self.eden_proc
+        cliques = defaultdict(int)
+        emb = eden_proc.tightestNativeBiClique(8, 12, chain_imbalance=None)
+        verify_biclique(eden_proc, emb, 8, 12, 3, 2)
```

Comment:
I'm confused. The test, `test_biclique_tightest(..)` appears twice. Once in lines 177-181 and a second time in lines 189-193.

Reply:
That's a bug. Nice catch.